crates/jstz_core/src/kv/outbox.rs

```diff
@@ -0,0 +1,461 @@
+use crate::error::Result;
+use derive_more::{Display, Error, From};
+use serde::{Deserialize, Serialize};
+use tezos_smart_rollup::{
+    core_unsafe::MAX_OUTPUT_SIZE,
+    michelson::{ticket::FA2_1Ticket, MichelsonContract, MichelsonPair},
+    outbox::{
+        AtomicBatch, OutboxMessageFull, OutboxMessageTransaction,
+        OutboxQueue as KernelSdkOutboxQueue, OUTBOX_QUEUE,
+    },
+    prelude::debug_msg,
+};
+
+use tezos_data_encoding::{enc::BinWriter, encoding::HasEncoding, nom::NomReader};
+use tezos_smart_rollup_host::{path::RefPath, runtime::Runtime};
+
+use super::Storage;
+
+/// Exposing tezos_smart_rollup::outbox::OUTBOX_QUEUE_ROOT
+const ROLLUP_OUTBOX_QUEUE_ROOT: RefPath = RefPath::assert_from(b"/__sdk/outbox");
+
+const OUTBOX_QUEUE_ROOT: RefPath = RefPath::assert_from(b"/outbox");
+
+type RollupOutboxQueue = KernelSdkOutboxQueue<'static, RefPath<'static>>;
+
+type NativeWithdrawalParameters = MichelsonPair<MichelsonContract, FA2_1Ticket>;
+
+// TODO: Might need to use OutboxMessageTransactionBatch for L1 encoding
```

Comment:
> How should the user define a set of outbox messages that are executed atomically?
It depends on the API expose to the user. In the PR, the outbox only exposes the native withdraw which is ok to just be a single message. Down the road, we could could make use of batch messages depending on the use case.
> How should/can the user handle errors that result from outbox messages?
Since outbox message application is a completely decoupled step, I think the way to go is for the user to have use an external setup to react to the errors

Reply:
Hmm we would need an oracle to watch for outbox message execution and the suspension would be for 2 weeks. But would be cool because  we could essentially revert the effects of failed application. For eg. if outbox withdrawal operation failed, then the protocol could restore the user's balance. 

Maybe an easier solution is to ensure that the withdrawal will always succeed? then the user can assume that a successful write to the outbox is as good as gold.

Reply:
That is definitely a nicer approach, completely forgot that we'd need to suspend it for 2 weeks 😬 😅